arxiv/util/tests/test_serialize.py

```diff
@@ -0,0 +1,128 @@
+"""Tests for :mod:`.serialize`."""
+
+from unittest import TestCase, mock
+from datetime import datetime, date
+from pytz import UTC, timezone
+from .. import serialize
+
+ET = timezone('US/Eastern')
+
+
+class TestISO8601JSONEncoder(TestCase):
+    """
+    Tests :func:`.serialize.dumps` and :class:`.serialize.ISO8601JSONEncoder`.
+    """
+
+    def test_encode_with_date(self):
+        """Encode JSON with a :class:`.date` object."""
+        self.assertEqual(
+            serialize.dumps(date(year=1993, month=11, day=3)),
+            '"1993-11-03"'
+        )
+        self.assertEqual(
+            serialize.dumps({"date": date(year=1993, month=11, day=3)}),
+            '{"date": "1993-11-03"}'
+        )
+        self.assertEqual(
+            serialize.dumps([{"date": date(year=1993, month=11, day=3)}]),
+            '[{"date": "1993-11-03"}]'
+        )
+        self.assertEqual(
+            serialize.dumps([{"date": [date(year=1993, month=11, day=3)]}]),
+            '[{"date": ["1993-11-03"]}]'
+        )
+
+    def test_encode_with_datetime(self):
+        """Encode JSON with a :class:`.datetime` object."""
+        self.assertEqual(
+            serialize.dumps(datetime(year=1993, month=11, day=3)),
+            '"1993-11-03T00:00:00"'
+        )
+        self.assertEqual(
+            serialize.dumps({"date": datetime(year=1993, month=11, day=3)}),
+            '{"date": "1993-11-03T00:00:00"}'
+        )
+        self.assertEqual(
+            serialize.dumps([{"date": datetime(year=1993, month=11, day=3)}]),
+            '[{"date": "1993-11-03T00:00:00"}]'
+        )
+        self.assertEqual(
+            serialize.dumps([{"date": [datetime(year=1993, month=11, day=3)]}]),
+            '[{"date": ["1993-11-03T00:00:00"]}]'
+        )
+
+    def test_encode_with_datetime_tz(self):
+        """Encode JSON with a TZ-aware :class:`.datetime` object."""
```

Comment:
Thanks for including tests with `tzinfo` set; it was the first thing I was looking for!